traffic_ops/testing/api/v2/deliveryservices_test.go

```diff
@@ -0,0 +1,602 @@
+
+	// DELETE+POST 1.1 again, so that 1.3 and 1.4 fields are back to nil/default
+	if _, err = TOSession.DeleteDeliveryService(strconv.Itoa(*respID)); err != nil {
+		t.Errorf("cannot DELETE deliveryservice: %s", err.Error())
+	}
+	postDSResp = tc.CreateDeliveryServiceNullableResponse{}
+	if err = makeV11Request(http.MethodPost, "deliveryservices", bytes.NewBuffer(dsV11Body), &postDSResp); err != nil {
+		t.Errorf("cannot POST 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	respID = postDSResp.Response[0].ID
+
+	// PUT 1.1 w/ 1.4 data - make sure 1.3 and 1.4 fields were ignored
+	putDSResp = tc.UpdateDeliveryServiceNullableResponse{}
+	if err = makeV11Request(http.MethodPut, "deliveryservices/"+strconv.Itoa(*respID), bytes.NewBuffer(dsBody), &putDSResp); err != nil {
+		t.Errorf("cannot PUT 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV13FieldsAreNil(putDSResp.Response[0]) || !dsV14FieldsAreNil(putDSResp.Response[0]) {
+		t.Errorf("PUT 1.1 expected 1.3 and 1.4 values to be nil, actual: non-nil %+v", putDSResp.Response[0])
+	}
+	respID = putDSResp.Response[0].ID
+	getDS, _, err = TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if !dsV13FieldsAreNilOrDefault(*getDS) || !dsV14FieldsAreNilOrDefault(*getDS) {
+		t.Errorf("PUT 1.1 expected 1.3 and 1.4 values to be nil/default, actual: non-nil/default %+v", *getDS)
+	}
+
+	// PUT 1.3 w/ 1.4 data, make sure 1.4 fields were ignored
+	putDSResp = tc.UpdateDeliveryServiceNullableResponse{}
+	if err = makeV13Request(http.MethodPut, "deliveryservices/"+strconv.Itoa(*respID), bytes.NewBuffer(dsBody), &putDSResp); err != nil {
+		t.Errorf("cannot PUT 1.1 deliveryservice, failed to make request: %s", err.Error())
+	}
+	if !dsV14FieldsAreNil(putDSResp.Response[0]) {
+		t.Error("PUT 1.3 expected 1.4 values to be nil, actual: non-nil")
+	}
+	respID = putDSResp.Response[0].ID
+	getDS, _, err = TOSession.GetDeliveryServiceNullable(strconv.Itoa(*respID))
+	if err != nil {
+		t.Errorf("cannot GET deliveryservice: %s", err.Error())
+	}
+	if !dsV14FieldsAreNilOrDefault(*getDS) {
+		t.Error("PUT 1.3 expected 1.4 values to be nil/default, actual: non-nil/default")
+	}
+}
+
+func dsV13FieldsAreNilOrDefault(ds tc.DeliveryServiceNullable) bool {
+	return (ds.DeepCachingType == nil || *ds.DeepCachingType == tc.DeepCachingTypeNever) &&
+		(ds.FQPacingRate == nil || *ds.FQPacingRate == 0) &&
+		(ds.TRRequestHeaders == nil || *ds.TRRequestHeaders == "") &&
+		(ds.TRResponseHeaders == nil || *ds.TRResponseHeaders == "")
+}
+
+func dsV14FieldsAreNilOrDefault(ds tc.DeliveryServiceNullable) bool {
+	return (ds.ConsistentHashRegex == nil || *ds.ConsistentHashRegex == "") &&
+		(ds.ConsistentHashQueryParams == nil || len(ds.ConsistentHashQueryParams) == 0) &&
+		(ds.MaxOriginConnections == nil || *ds.MaxOriginConnections == 0)
+}
+
+func dsV13FieldsAreNil(ds tc.DeliveryServiceNullable) bool {
```

Comment:
This function should serve no purpose assume the above revisions are made - so it can be removed.